CONTRIBUTING.md

```diff
@@ -0,0 +1,247 @@
+  renaming support to pyls.
+
+With these packages installed and a correctly setup editor (see the readme and
+feel free to reach out to us), your code should automatically follow all the
+standards which we could automate.
+
+The final sections, `tool.black`, `tool.isort`, `build-system`, and the
+`.flake8` file (Flake8 does not yet support `pyproject.toml`) configure the
+tools per their recommendations.
+
+## Naming Conventions
+
+One exception to the automation is enforcing naming standards. The [naming
+conventions](https://www.python.org/dev/peps/pep-0008/#naming-conventions)
+section of PEP 8 describes what each of the different styles means. A short
+summary of the most important parts:
+
+* Modules should have short, all-lowercase names.
+* Class (and exception) names should normally use the `CapWords` convention
+  (also known as `CamelCase`).
+* Function and variable names should be lowercase, with words separated by
+  underscores as necessary to improve readability (also known as `snake_case`).
+* To avoid collisions with the standard library, an underscore can be appended,
+  such as `id_`.
+* Always use `self` for the first argument to instance methods.
+* Always use `cls` for the first argument to class methods.
+* Use one leading underscore only for non-public methods and instance variables,
+  such as `_data`.
+* Constants should be `CAPITALIZED_SNAKE_CASE`.
+
+When in doubt, adhere to existing conventions, or check the style guide.
+
+## Type Annotations
+
+We are using [mypy][] to enforce static type checking of our Python code. This
+may surprise you as Python is not a statically typed language. While dynamic
+typing can be useful, for a complex tool such as LISA it is more likely to
+introduce bugs that are found only at runtime (which the user experiences as a
+crash). For more information on why we (and others) do this, see [Dropbox’s
+journey to type checking 4 million lines of Python][dropbox]. [PEP 484][] and
+[PEP 526][] (among others) introduced and defined [type hints][] for the Python
+language. You can probably figuring out the syntax based on the surrounding
+code, but you can also see this [Intro to Using Python Type Hints][intro] and
+mypy’s [cheat sheet][].
+
+[mypy]: http://mypy-lang.org/
+[dropbox]: https://dropbox.tech/application/our-journey-to-type-checking-4-million-lines-of-python
+[PEP 484]: https://www.python.org/dev/peps/pep-0484/
+[PEP 526]: https://www.python.org/dev/peps/pep-0526/
+[type hints]: https://docs.python.org/3/library/typing.html
+[intro]: https://kishstats.com/python/2019/01/07/python-type-hinting.html
+[cheat sheet]: https://mypy.readthedocs.io/en/latest/cheat_sheet_py3.html
+
+## Committing Guidelines
+
+A best practice when using [Git](https://git-scm.com/book/en/v2) is to create a
+series of independent and well-documented commits. Each commit should “do one
+thing” and do it correctly. If a mistake is made (you need to fix a bug or
```

Comment:
One thing I'd mention are the oft-unknown duo: `git add -p` and `git reset -p`, which will, respectively, add which changes to stage (allowing you even open a text editor to make changes before staging them), or remove a particular change from the stage.

They're very useful when you end up fixing an unrelated bug or typo and don't want to include them in the same commit you're introducing a new feature or implementing something else.

Reply:
These are...unknown to people?! How do they use Git otherwise? 😅

Reply:
There are a LOT of Git tips and tricks we can add. I'll do this in another iteration. Thanks @lpereira!

Reply:
git stash can archive more untracked changes. When using git rebase, it don't allow any local change, git stash can help.